libs/langchain/langchain/document_loaders/pdf.py

```diff
@@ -66,10 +66,11 @@ class BasePDFLoader(BaseLoader, ABC):
     to a temporary file, use it, then clean up the temporary file after completion
     """
 
-    def __init__(self, file_path: str):
+    def __init__(self, file_path: str, file_headers: Optional[Dict] = None):
```

Comment:
Thanks for the contribution! 

Let's call these headers to it's not confused with file signatures (https://en.wikipedia.org/wiki/List_of_file_signatures)

```suggestion
    def __init__(self, file_path: str, *, headers: Optional[Dict] = None):
```
Could you expand the initilaizer doc-string and document both arguments?

Reply:
@eyurtsev 
MathpixPDFLoader that inherit BasePDFLoader already has a property (line:355) with name 'header'. so there is likely to be a conflict when using it.

Or would it be okay to change the 'headers' property of MathpixPDFLoader to 'api_headers'?? It would be a good idea to change MathpixPDFLoader's 'url' and 'data' properties to 'api_url' and 'api_data'.

Reply:
Good idea